packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -887,3 +889,40 @@ class HeroController extends NavigatorObserver {
     return toHero.child;
   };
 }
+
+/// Enables or disables [Hero]es in the widget subtree.
+class HeroMode extends StatelessWidget {
```

Comment:
This could inherit from ProxyWidget instead?

Reply:
ProxyWidget is abstract, the code addition will increase. SingleChildRenderObjectWidget is good for this, right?

Reply:
Is it better to inherit ProxyWidget and return SingleChildRenderObjectElement?

Reply:
No, SingleChildRenderObjectWidget is expected to have a renderObject, The HeroMode itself does not layout or paint in the render layer. It is only adding information to the widget tree. The ProxyWidget returns ComponentElement might make most sense in this case.

Note: there is a ProxyElement, but it has specific logic built in to notify child when its configuration changes. We do not need this functionality, so i think ProxyElement is not suitable in this use case

Reply:
This seems not be addressed yet?